### PR TITLE
Fix missing strict types for the social service

### PR DESCRIPTION
**Pull request by ChristophWurst**

### CODE REVIEW

**Comment by call-me-matt** [comment]
I guess we have to convert the http status to string [here](https://github.com/nextcloud/contacts/blob/f6603af4109d61eeab9ce586ec83e433df73cfd0/tests/unit/Service/SocialApiServiceTest.php#L112) for unit tests to pass with strict types

**Review by skjnldsv — APPROVED** [review]

**Comment by ChristophWurst** [comment]
> I guess we have to convert the http status to string [here](https://github.com/nextcloud/contacts/blob/f6603af4109d61eeab9ce586ec83e433df73cfd0/tests/unit/Service/SocialApiServiceTest.php#L112) for unit tests to pass with strict types
> ... or to integer [here](https://github.com/nextcloud/contacts/blob/f6603af4109d61eeab9ce586ec83e433df73cfd0/lib/Service/SocialApiService.php#L292)?

It should be an int. Can you fix this please?

**Review by miaulalala — APPROVED** [review]

File: lib/Service/SocialApiService.php [hunk 1]
```diff
@@ -408,9 +411,9 @@ public function updateAddressbooks(string $userId, string $offsetBook = null, st
 
 				try {
 					$r = $this->updateContact($addressBook->getURI(), $contact['UID'], $network);
-					$response = $this->registerUpdateResult($response, $contact['FN'], $r->getStatus());
+					$response = $this->registerUpdateResult($response, $contact['FN'], int($r->getStatus()));
```
Comment (ChristophWurst) [inline lib/Service/SocialApiService.php:414]:
```suggestion
					$response = $this->registerUpdateResult($response, $contact['FN'], (int) $r->getStatus());
```

Comment (call-me-matt) [reply lib/Service/SocialApiService.php:414]: ah yes, sorry I used your pipeline for testing. Didn't have a dev setup.

**Review by call-me-matt — APPROVED** [review]

**Review by aloxe — APPROVED** [review]

**Pull request merged**